skimage/restoration/_cycle_spin.py

```diff
@@ -34,44 +34,18 @@ def _generate_shifts(ndim, multichannel, max_shifts, shift_steps=1):
     elif len(shift_steps) != ndim:
         raise ValueError("max_shifts should have length ndim")
 
-    if np.any(np.asarray(shift_steps) < 1):
+    if any(s < 1 for s in shift_steps):
```

Comment:
Is it faster?

Reply:
It is faster, but we are talking differences in microseconds so it is unlikely to make a practical performance difference in applications. The calls to `func` within `cycle_spinning` are likely to take orders of magnitude longer than the call to this `_generate_shifts` function.

example:
```
shift_steps = (1, 1, 0)

%timeit np.any(np.asarray(shift_steps) < 1)
18.9 µs ± 4.22 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)

%timeit any(s < 1 for s in shift_steps)
581 ns ± 49.7 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

Reply:
actually, I will qualify that it is faster in the specific case that shift_steps is small (which will always be the case here because the length of `shift_steps` is equal to the number of data dimensions).

If there were millions of elements, numpy would be faster in general (but still slower in a case where there is a value < 1 near the start of the array)